File: packages/ios/Sources/CivUI/CivActionLink.swift
// CivUI — CivActionLink for SwiftUI
// Unified action link for phone calls and email compose.
// Renders the correct protocol, icon, and formatted display text based on type.

import SwiftUI

/// Action link type — determines protocol and icon.
public enum CivActionLinkType: String {
    case phone
    case email
}

/// Unified action link for device actions (phone call, email compose).
///
/// Usage:
/// ```swift
/// CivActionLink(type: .phone, number: "8005551234")
/// CivActionLink(type: .email, address: "help@va.gov", subject: "Question")
/// ```
public struct CivActionLink: View {
    public var type: CivActionLinkType
    public var number: String
    public var address: String
    public var subject: String
    public var label: String
    public var disabled: Bool

    public init(
        type: CivActionLinkType = .phone,
        number: String = "",
        address: String = "",
        subject: String = "",
        label: String = "",
        disabled: Bool = false
    ) {
        self.type = type
        self.number = number
        self.address = address
        self.subject = subject
        self.label = label
        self.disabled = disabled
    }

    public var body: some View {
        EmptyView() // TODO: Implement with CivLink delegation
    }
}

#if DEBUG
struct CivActionLink_Previews: PreviewProvider {
    static var previews: some View {
        VStack(alignment: .leading, spacing: 12) {
            CivActionLink(type: .phone, number: "8005551234")
            CivActionLink(type: .email, address: "help@va.gov")
        }
        .padding()
    }
}
#endif

File: packages/android/src/main/kotlin/gov/civui/components/CivActionLink.kt
// CivUI — CivActionLink for Jetpack Compose
// Unified action link for phone calls and email compose.
// Renders the correct protocol, icon, and formatted display text based on type.

package gov.civui.components

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview

/**
 * Unified action link for device actions (phone call, email compose).
 *
 * Usage:
 * ```kotlin
 * CivActionLink(type = "phone", number = "8005551234")
 * CivActionLink(type = "email", address = "help@va.gov", subject = "Question")
 * ```
 */
@Composable
fun CivActionLink(
    type: String = "phone",
    number: String = "",
    address: String = "",
    subject: String = "",
    label: String = "",
    disabled: Boolean = false,
    modifier: Modifier = Modifier,
) {
    // TODO: Implement with CivLink delegation
}

@Preview(showBackground = true, name = "CivActionLink")
@Composable
private fun CivActionLinkPreview() {
    CivActionLink(type = "phone", number = "8005551234")
}
